sdk/core/azure-core/src/main/java/com/azure/core/http/rest/ResponseExceptionConstructorCache.java

```diff
@@ -40,18 +40,30 @@ private static MethodHandle locateExceptionConstructor(Class<? extends HttpRespo
                 exceptionBodyType);
 
             return lookupToUse.unreflectConstructor(constructor);
-        } catch (Throwable t) {
-            throw LOGGER.logExceptionAsError(new RuntimeException(t));
+        } catch (Exception ex) {
+            if (ex instanceof RuntimeException) {
+                throw LOGGER.logExceptionAsError((RuntimeException) ex);
+            }
+
+            throw LOGGER.logExceptionAsError(new RuntimeException(ex));
```

Comment:
I wonder if we should add `RuntimeException logExceptionAsErrorAndPropagate(Exception e)` or overload `RuntimeException logExceptionAsError(Exception e)` to reduce verbosity.
But then we we'd need something for Throwables and it's going to be hard to find something to capture both `RuntimeException` and `Error` as return type. So maybe `LOGGER.logAndPropagate(Throwable)` that does the throw inside ? That'd be one extra line in stack trace but other than that same information preserved.

Reply:
I wonder if a method like `extractOrCreateRuntimeException(Throwable, Function<Throwable, RuntimeException>)` would be better, I'm not sure I'd want to merge extracting and logging into a single utility method as that may result in a proliferation to handle different cases.

Reply:
other option is to write checkstyle rule that enforces certain styles of try-catch blocks if `Throwable` or `Exception` are getting caught.
Btw. we can use multiple catches instead of instanceof in those templates. for example:

```
} catch (RuntimeException e) {
   LOGGER.log...(e);
} catch (Exception e) {
   LOGGER.log(wrap(e));
} 
```
or if we're dealing with api that throws throwables

```
} catch (RuntimeException e) {
   LOGGER.log...(e);
} catch (Exception e) {
   LOGGER.log(wrap(e));
} catch (Throwable t) {
 throw (Error) t
}
```

Reply:
Feel free to punt this from this PR.